### PR TITLE
Add variables to support TLS version and cipher in the OTLP exporter

### LINKED ISSUE #2932: Need new options to specify minimum TLS version and allowed ciphers list
I am looking for a way how to configure TLS options for OTLP HTTP exporter. I found that I can use https schema and set path to my trusted CA certificate file using exporter option or `OTEL_EXPORTER_OTLP_CERTIFICATE` environment variable, as specified here:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md

I also need to specify TLS 1.2 as a minimum TLS version and list of allowed ciphers to use. There are no options to set them now. Please add them to the specification.

Note: Please make sure that OpenTelemetry Collector also would allow to configure these two options for its importers and exporters. I am not sure if it uses the same specification or or has its own one.

### PR DESCRIPTION
**Pull request by marcalff**
Fixes #2932

## Changes

Add environment variables to support:

* min TLS version
* max TLS version
* TLS cipher list (for TLS <= 1.2)
*  TLS cipher suite (for TLS 1.3)

in the OTLP EXPORTER configuration.

### CODE REVIEW

**Comment by marcalff** [comment]
To @open-telemetry/specs-approvers ,

Any comments on this PR ?

I am not even sure if it was noticed at all.

Regards.

**Comment by marcalff** [comment]
So, this PR was closed automatically by a bot, without any human reviewer actually looking at it.

The contribution process is severely flawed in my opinion.

The _author_ is not to blame for _lack of activity_ here.

**Comment by yurishkuro** [comment]
@marcalff I understand it can be upsetting when a PR is not being merged, but as a project we have priorities and limited resources, which are directed at things that people find important. It does not look like the feature requested in the ticket is in high demand, or supported by any SDKs / collector.

**Comment by marcalff** [comment]
> @marcalff I understand it can be upsetting when a PR is not being merged, but as a project we have priorities and limited resources, which are directed at things that people find important. It does not look like the feature requested in the ticket is in high demand, or supported by any SDKs / collector.

@yurishkuro Thanks for taking the time to comment.

I know full well people have limited resources, facing the same issues in opentelemetry-cpp.

The friction here is not so much about the PR being merged or not, but about the lack of any comments, even negative.

Had anyone asked, this feature is supported in the opentelemetry-cpp SDK, per PR https://github.com/open-telemetry/opentelemetry-cpp/pull/1793, which will not be merged then, since the spec is rejected.

**Comment by tigrannajaryan** [comment]
@marcalff has anything changed since this was closed? I can reopen but it won't be of much use if no-one reviews/comments on it and it is closed again. To gain some traction I think it is best to find some other people who are interested in this and ask them to review and comment (ideally it would be spec approvers, but can also be any other Otel members). Also advertise in the specification SIG meeting and maintainers' meeting, in CNCF Slack channels and with any other relevant audiences.

**Comment by marcalff** [comment]
@tigrannajaryan Thanks for taking the time to comment.

Nothing has changed indeed.

**Comment by svrnm** [comment]
It looks like the underlying issue and by that this PR became important once again based on [this discussion](https://cloud-native.slack.com/archives/C01NJ7V1KRC/p1718207166293719) by @jpkrohling

File: specification/protocol/exporter.md [hunk 1]
```diff
@@ -32,6 +32,36 @@ The following configuration options MUST be available to configure the OTLP expo
   - Default: n/a
   - Env vars: `OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE` `OTEL_EXPORTER_OTLP_TRACES_CLIENT_CERTIFICATE` `OTEL_EXPORTER_OTLP_METRICS_CLIENT_CERTIFICATE` `OTEL_EXPORTER_OTLP_LOGS_CLIENT_CERTIFICATE`
 
+- **TLS minimum version**: Minimum TLS version to use with SSL.
+  Should only be used for a secure connection.
+  See [TLS version](./exporter.md#tls-version) for more details.
+  - Default: n/a
```
Comment (svrnm) [inline specification/protocol/exporter.md:38]: It looks like this should be set to 1.3 (if supported?)

**Comment by marcalff** [comment]
> It looks like the underlying issue and by that this PR became important once again based on [this discussion](https://cloud-native.slack.com/archives/C01NJ7V1KRC/p1718207166293719) by @jpkrohling

This is actually the discussion that made me look at this old PR again.

I think that at some point, and in particular to graduate opentelemetry in general, we will need to cover security much more extensively that we do today.

This assessment however, does not seem to be widely accepted yet.

**Comment by trask** [comment]
hi @marcalff! are you able to join the Specification SIG meeting and drive some discussion around this PR? it's not uncommon for specification issues or PRs to go stale or be overlooked, and sometimes (🤞) all it takes is 10 minutes of synchronous discussion in the Specification SIG meeting to unblock or gather interest in it

**Comment by marcalff** [comment]
Hi @trask 

From what I could find, the spec SIG meeting is at: Tue at 8:00am Pacific Time

I will attend the meeting on Tuesday June 25, 2024 then (unable to attend next week).

**Comment by trask** [comment]
Tuesday June 25 is [OpenTelemetry Community Day](https://events.linuxfoundation.org/open-telemetry-community-day/), attendance will probably be very light (or canceled), so I'd recommend joining a future meeting instead

**Comment by marcalff** [comment]
Related:

https://github.com/open-telemetry/opentelemetry-configuration/issues/322

**Pull request closed without merging**